frame/election-provider-support/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Election provider support pallet benchmarking.
+//! This is separated into its own crate to avoid bloating the size of the runtime.
+
+#![cfg(feature = "runtime-benchmarks")]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::Decode;
+use frame_benchmarking::{benchmarks, Vec};
+use frame_election_provider_support::{NposSolver, PhragMMS, SequentialPhragmen};
+
+pub struct Pallet<T: Config>(frame_system::Pallet<T>);
+pub trait Config: frame_system::Config {}
```

Comment:
The issues are:
- The benchmark system seems to impose that the `Config` trait is implemented for `Runtime`, however the `onchain::Config` is not (and by design because otherwise we cannot have 2 different configs for `Runtime` for example the 2 algorithms).
- Don't we ideally want to have the benchmarking bounds different from the trait bounds? For example say the pallet is unbounded (bounded by the max value), we would still want to cap the benchmarking to not take a lot of time to run.
- I thought about adding `BenchmarkingConfig` to configure that, but then we need `BenchmarkingConfig` to be implemented in the node runtime because each chain may want to benchmark it differently, and we need the benchmarking pallet to have access to it, so it probably needs to be implemented for `Runtime`, this looked like over engineering it for now, and having these values defined as constant in the pallet itself acceptable.

Obviously happy to take suggestions to the contrary.

Reply:
Okay, I see the problem. For now let's keep it as-is, and keep the benchmarking configs "constant", and later on we can add them to this `trait Config` if needed.